packages/x-tree-view/src/RichTreeView/RichTreeView.tsx

```diff
@@ -0,0 +1,323 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { styled, useThemeProps } from '@mui/material/styles';
+import composeClasses from '@mui/utils/composeClasses';
+import { useSlotProps } from '@mui/base/utils';
+import { getRichTreeViewUtilityClass } from './richTreeViewClasses';
+import { RichTreeViewProps } from './RichTreeView.types';
+import { useTreeView } from '../internals/useTreeView';
+import { TreeViewProvider } from '../internals/TreeViewProvider';
+import { DEFAULT_TREE_VIEW_PLUGINS } from '../internals/plugins';
+import { TreeItem, TreeItemProps } from '../TreeItem';
+import { buildWarning } from '../internals/utils/warning';
+
+const useUtilityClasses = <R extends {}, Multiple extends boolean | undefined>(
+  ownerState: RichTreeViewProps<R, Multiple>,
+) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getRichTreeViewUtilityClass, classes);
+};
+
+export const RichTreeViewRoot = styled('ul', {
+  name: 'MuiRichTreeView',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: RichTreeViewProps<any, any> }>({
+  padding: 0,
+  margin: 0,
+  listStyle: 'none',
+  outline: 0,
+});
+
+type TreeViewComponent = (<R extends {}, Multiple extends boolean | undefined = undefined>(
+  props: RichTreeViewProps<R, Multiple> & React.RefAttributes<HTMLUListElement>,
+) => React.JSX.Element) & { propTypes?: any };
+
+function WrappedTreeItem<R extends {}>({
+  slots,
+  slotProps,
+  label,
+  id,
+  nodeId,
+  children,
+}: Pick<RichTreeViewProps<R, any>, 'slots' | 'slotProps'> &
+  Pick<TreeItemProps, 'id' | 'nodeId' | 'children'> & { label: string }) {
+  const Item = slots?.item ?? TreeItem;
+  const itemProps = useSlotProps({
+    elementType: Item,
+    externalSlotProps: slotProps?.item,
+    additionalProps: { nodeId, id, label },
+    ownerState: { nodeId, label },
+  });
+
+  return <Item {...itemProps}>{children}</Item>;
+}
+
+const childrenWarning = buildWarning([
+  'The `RichTreeView` component does not support JSX children.',
```

Comment:
WDYT about at least prefixing the warning with `MUI: `? 🤔 
```suggestion
  'MUI: The `RichTreeView` component does not support JSX children.',
```